packages/cursorless-engine/src/processTargets/modifiers/getContainingScopeTarget.ts

```diff
@@ -0,0 +1,149 @@
+import { Direction, Position, TextEditor } from "@cursorless/common";
+import type { Target } from "../../typings/target.types";
+import { constructScopeRangeTarget } from "./constructScopeRangeTarget";
+import { getContainingScope } from "./getContainingScope";
+import { TargetScope } from "./scopeHandlers/scope.types";
+import { ScopeHandler } from "./scopeHandlers/scopeHandler.types";
+
+/**
+ * Finds the containing scope of the target for the given scope handler
+ *
+ * @param target The target to find the containing scope of
+ * @param scopeHandler The scope handler for the scope type to find containing scope of
+ * @param ancestorIndex How many ancestors to go up. 0 means the immediate containing scope
+ * @returns A target representing the containing scope, or undefined if no containing scope found
+ */
+export function getContainingScopeTarget(
```

Comment:
Factored out the core implementation of `ContainingScopeStage` so that `EveryScopeStage` could use it directly for expanding to containing iteration scope, because iteration scopes don't necessarily have a well-defined scope type, so it's easier if it can just pass in a scope handler rather than making a dummy scope type just for this purpose